### ISSUE TITLE
Consolidate configuration to only runtime configuration + default artifacts

### ISSUE THREAD

**Issue by nholland94** [CodaProtocol/coda#6013]
Our code currently supports runtime configuration overrides with constant time defaults. When handling lack of explicit runtime configuration overrides, the daemon falls back to constant time defaults. When deciding to generate configuration dependent artifacts (i.e. genesis ledger, pk/vk keys, genesis proof, genesis block), the daemon has a variety of ways to lookup cached artifacts before it generates anything. These artifacts can exist in different locations, depending on which artifact is being generated. These locations can include: compiled into the daemon, locally cached on the file system, locally distributed along with the daemon, available for download from s3. The current initialization behavior is thus rather complex, and can manifest a variety of code paths for selecting what data to initialize the daemon with. We should consolidate these code paths into as few as possible, making the logic easier to reason about, the daemon easier to configure, the initialization easier to debug, and the implementation of hard forks easier to test and develop.

Talking with @bkase and @mrmr1993, it seems that we would like the primary means of loading configuration dependent artifacts to be via the local filesystem. This allows us to dynamically regenerate artifacts when configuration changes without the need to ship a new daemon, but it still allows us to provide some artifacts that can be used with a default configuration (important for shipping software at mainnet). We can still rely on s3 in the short term for pv/vk keys, but the ideal long term solution would be to have our deployments bundle the pv/vk keys into the distribution it wants to deploy, so the s3 cache logic can eventually be removed from the daemon itself (and same goes for other artifacts too).

**Comment by lk86** [comment]
I totally agree with this approach. Bundling artifacts alongside deb + brew packages in /var/lib/coda like we have done for v3.3 is both easy to debug/understand and easy to control / roll out changes. The s3 caching as it exists now causes both a high AWS cost and slow/inconsistent startup procedures and a lot of logic within the daemon that we don't need. Artifacts should remain in the brew/deb logic but should be removed from the daemon code itself.

**Comment by psteckler** [comment]
+1 for simpler config loading.

**Comment by mobileink** [comment]
What do you mean by "constant time defaults"?  Compile time?  That covers two cases:

- hard-coded constants
- arguments to the compiler
  - via command line
  - from a config file

Actually there's a third case, hard-coded constants that are set by the build commands, e.g. -DDEBUG

I ask because Bazel needs to know.

**Comment by nholland94** [comment]
All the compile-time constants we are referring to in this case live in the `src/config/*.mlh` files, and are fed into the OCaml compile via a ppx `ppx_optcomp`.
